src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/Registration/RegistrationExistsValidator.cs

```diff
@@ -30,6 +30,15 @@ public override Task<bool> ShouldRunLeaf(ValidationContext data, PackageRegistra
 
             if (v2Exists != v3Exists)
             {
+                // Currently, leaf nodes are not deleted after a package is deleted.
+                // This is a known bug. Do not fail validations because of it.
+                // See https://github.com/NuGet/NuGetGallery/issues/4475
+                if (!v2Exists && v3Exists && !(v2 is PackageRegistrationIndexMetadata) && !(v3 is PackageRegistrationIndexMetadata))
```

Comment:
Why are we checking types here? Could you explain the meaning here?

Reply:
When we are running validations against an index here, the type of `v2` and `v3` will be `PackageRegistrationIndexMetadata`. If we are running validations against a leaf, it will be a `PackageRegistrationLeafMetadata`, which `PackageRegistrationIndexMetadata` inherits from. Therefore, we are checking that `v3` is not a `PackageRegistrationIndexMetadata` because we only want to avoid failing validations if it is a `PackageRegistrationLeafMetadata`.

Reply:
From what I gather, there would be only two types of metadata, 'index' and 'leaf, wouldn't this be simplified by:
```
if (v3Exists && !(v3 is PackageRegistrationIndexMetadata))
```
The third check `(!(v2 is PackageRegistrationIndexMetadata))` might altogether be unnecessary, because if it doesn't exist, wouldn't it be null?

Reply:
Correct. I will fix.